content/blog/deploying-netlify-cms-on-aws/index.md

```diff
@@ -0,0 +1,363 @@
+
+![Change Repo Name](./change-repo-name.jpg)
+
+### File Structure
+
+Simplifying the file and deleting unneeded parameters produces this:
+
+![File Structure](./file-structure.jpg)
+
+The `./src/index.js` file is the entry point for React to create the App class inside `./public/index.html`'s  `<div id=root>`. `App` (defined in `./src/App.js`) is a React Component that returns the `NetlifyCMS` React component defined inside `./src/components/NetlifyCMS/index.js`. We can also specify the [custom preview](https://www.netlifycms.org/docs/customization/#registerpreviewtemplate) and [custom widget](https://www.netlifycms.org/docs/custom-widgets/#header) for the CMS app in this file.
+
+To preview the CMS app content, run `yarn start`.
+
+Now that we have preprocessed the file structure. Let's deploy the CMS onto the AWS!
+
+## Implement Infrastructure
+
+First, we create an `infrastructure` folder to store the Pulumi project. To create a new Pulumi project, run `pulumi new typescript` to generate a typescript template.
+
+We can use Pulumi's [static website example](https://github.com/pulumi/examples/tree/master/aws-ts-static-website) as a model for deploying our application.
+
+### Step 1: Build the CMS App
+
+We need to build the website with `yarn build`. This creates a `build` folder under the root directory that holds the compiled version of the website contents to be uploaded to an S3 bucket. Once uploaded, the S3 bucket contains a working version of our CMS app, which reduces the number of files to upload.
+
+### Step 2: Fetch Configuration
+
+The first step is to create a stack configuration with our stack name.
+
+```typescript
+// Replace "pulumi-website-cms" with your stack name
+const stackConfig = new pulumi.Config("pulumi-website-cms");
+const config = {
+    pathToWebsiteContents: stackConfig.require("pathToWebsiteContents"),
+    targetDomain: stackConfig.require("targetDomain"),
+    certificateArn: stackConfig.get("certificateArn"),
+};
+```
+
+The configuration takes three parameters set with the Pulumi CLI
+
+```bash
+$ pulumi config set pulumi-website-cms:pathToWebsiteContent ../build
+```
+
+The pathToWebsiteContent parameter specifies the path to the directory with the website content. This directory is synched to the S3 bucket by using the `crawlDirectory` method. We pass in the `build` folder, which is the directory that stores our built CMS website.
+
+Similarly, we can set the CMS's target domain name and the optional certificate arn parameter with `stackConfig.get` instead of `stackConfig.require` to get the value. For the target domain, we can use any subdomain name of an existing parent domain and the code creates the subdomain under the parent domain automatically. For the certificate arn, if we already have a certificate registered for CMS then we can pass it in as the certificate arn. Otherwise, the code generates a new certificate.
```

Comment:
```suggestion
Similarly, we can set the CMS's target domain name and the optional certificate arn parameter with `stackConfig.get` instead of `stackConfig.require` to get the value. For the target domain, we can use any subdomain name of an existing parent domain, and the code creates the subdomain under the parent domain automatically. For the certificate arn, if we already have a certificate registered for CMS, then we can pass it in as the certificate arn. Otherwise, the code generates a new certificate.
```